testing/xfpga/test_error_c.cpp

```diff
@@ -92,22 +92,34 @@ class error_c_p
   _fpga_token fake_port_token_;
 };
 
-int error_c_p::delete_errors(std::string fpga_type, std::string filename) {
+int error_c_mock_p::delete_errors(std::string fpga_type, std::string filename) {
   int result;
-  std::string fme_sysfspath = tmpsysfs_ + "/" + sysfs_fme + "/" + filename;
-  std::string port_sysfspath = tmpsysfs_ + "/" + sysfs_port + "/" + filename;
+  std::string fme_sysfspath;
+  std::string port_sysfspath; 
+  std::string cmd;
+ 
+  if (tmpsysfs_.length() < 2) {
+    fme_sysfspath = tmpsysfs_ + sysfs_fme + "/" + filename;
+    port_sysfspath = tmpsysfs_ + sysfs_port + "/" + filename;
+  }
+  else {
+    fme_sysfspath = tmpsysfs_ + "/" + sysfs_fme + "/" + filename;
+    port_sysfspath = tmpsysfs_ + "/" + sysfs_port + "/" + filename;
+  }
+
   if (fpga_type.compare("fme") == 0){
-    auto cmd = "rm -rf " + fme_sysfspath;
-    result = std::system(cmd.c_str());
-    (void) result;
-    return 1;
+    cmd = "rm -rf " + fme_sysfspath;
+    goto remove;
   } else if (fpga_type.compare("port") == 0){
-    auto cmd = "rm -rf " + port_sysfspath;
-    result = std::system(cmd.c_str());
-    (void) result;
-    return 1;
+    cmd = "rm -rf " + port_sysfspath;
+    goto remove;
   }
   else {return -1;}
+
+remove:
+  result = std::system(cmd.c_str());
```

Comment:
Can we use an EXPECT_EQ here to check the return value and remove result altogether?